client-tez/src/main/java/org/apache/tez/runtime/library/output/RssOrderedPartitionedKVOutput.java

```diff
@@ -212,6 +213,16 @@ public List<Event> close() throws Exception {
   public void start() throws Exception {
     if (!isStarted.get()) {
       memoryUpdateCallbackHandler.validateUpdateReceived();
+      int maxFailures =
+          conf.getInt(
+              TezConfiguration.TEZ_AM_TASK_MAX_FAILED_ATTEMPTS,
+              TezConfiguration.TEZ_AM_TASK_MAX_FAILED_ATTEMPTS_DEFAULT);
+      boolean speculation =
+          conf.getBoolean(
+              TezConfiguration.TEZ_AM_SPECULATION_ENABLED,
+              TezConfiguration.TEZ_AM_SPECULATION_ENABLED_DEFAULT);
+      int maxAttemptNo = ClientUtils.getMaxAttemptNo(maxFailures, speculation);
+      int rsstaskAttemptId = RssTezUtils.createRssTaskAttemptId(taskAttemptId, maxAttemptNo);
```

Comment:
This code is duplicated a few times, `RssTezUtils` should implement `createRssTaskAttemptId(taskAttemptId, maxAttemptNo, conf)`.